pandas/core/reshape/merge.py

```diff
@@ -1015,6 +1015,43 @@ def _get_join_info(
         left_ax = self.left.axes[self.axis]
         right_ax = self.right.axes[self.axis]
 
+        if self.how in ["anti_left", "anti_right", "anti_full"]:
+            how_dict: dict[str, str] = {
+                "anti_left": "left",
+                "anti_right": "right",
+                "anti_full": "outer",
+            }
+            self.how = how_dict[self.how]
+            if self.left_index and self.right_index:
+                # Merge using `right_index` and `left_index`
+                left_ax = self.left.index[~self.left.index.isin(self.right.index)]
```

Comment:
Curious if you've tried something closer to a left join with a filter on subsequent NA values compared to this. At first glance this seems like it could be a pretty expensive way of calculating this

Reply:
honestly I haven't tried that, I just went with https://github.com/pandas-dev/pandas/issues/42916#issuecomment-894430300 from @attack68

Reply:
For reference, below is the asv comparison

```python
[ 50.00%] · For pandas commit 139881dd <42916v2> (round 2/2):
[ 50.00%] ·· Benchmarking conda-py3.8-Cython0.29.32-jinja2-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pyarrow-pytables-scipy-sqlalchemy-xlrd-xlsxwriter-xlwt
[ 75.00%] ··· join_merge.I8Merge.time_i8merge                                                                                                                           ok
[ 75.00%] ··· ============ ============
                  how                  
              ------------ ------------
                 inner      1.13±0.03s 
                 outer      1.12±0.02s 
                  left      1.19±0.01s 
                 right      1.24±0.05s 
               anti_left     567±20ms  
               anti_right    572±20ms  
               anti_full     577±30ms  
              ============ ============
```

Reply:
IIUC this is the same operation as np.setdiff1d . Generally can you look at using the numpy functions here?

Reply:
are you suggesting this `numpy` thing specifically here in `_get_join_info `? I will have to try how that works out with something like EA / Categorical dtypes